homeassistant/components/iammeter/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow for iammeter integration."""
+import asyncio
+import logging
+from urllib.parse import ParseResult, urlparse
+
+import async_timeout
+import iammeter
+from iammeter.power_meter import IamMeterError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_TIMEOUT = 8
+
+
+@callback
+def iammeter_entries(hass: HomeAssistant):
+    """Return the hosts already configured."""
+    return {
+        entry.data[CONF_NAME] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class IammeterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Iammeter."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.host = None
+
+    def _host_in_configuration_exists(self, host) -> bool:
+        """Return True if host exists in configuration."""
+        if host in iammeter_entries(self.hass):
+            return True
+        return False
+
+    async def _test_connection(self, host, port):
+        with async_timeout.timeout(PLATFORM_TIMEOUT):
+            try:
+                await iammeter.real_time_api(host, port)
+                return True
+            except (IamMeterError, asyncio.TimeoutError) as err:
+                _LOGGER.error("Device is not ready!")
+                raise IamMeterError from err
+        return False
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes a integration."""
+        errors = {}
+        if user_input is not None:
+            # set some defaults in case we need to return to the form
+            name = user_input.get(CONF_NAME, DEFAULT_NAME)
+            port = user_input.get(CONF_PORT, DEFAULT_PORT)
+            host_entry = user_input.get(CONF_HOST)
+            url = urlparse(host_entry, "http")
+            netloc = url.netloc or url.path
+            path = url.path if url.netloc else ""
+            url = ParseResult("http", netloc, path, *url[3:])
+            host = netloc
+
+            if self._host_in_configuration_exists(name):
```

Comment:
The aimmeter API provides the serial number of the device.
Should that be used as the unique identifier for the configuration entry instead?

`api.iammeter.serial_number`

Reply:
Thansk for your advice,
I am about to add this function in the next PR(add ssdp auto find),this PR is enable user input their device name manually.

Reply:
Manually or not, the unique ID needs to be consistent. 
Since there is a consistent ID available, you should use that. For me, this needs to be changed in this PR, not the next PR.

Reply:
Using serial_number as uid now.
Thanks for your advice.

Reply:
@frenck would you mind review my codes when you have spare time,thanks a lot!